cryptoservice/crypto_service.go

```diff
@@ -58,16 +64,29 @@ func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (*dat
 	if err != nil {
 		return nil, fmt.Errorf("failed to add key to filestore: %v", err)
 	}
-	return data.PublicKeyFromPrivate(*privKey), nil
+	return data.PublicKeyFromPrivate(privKey), nil
+}
+
+// GetKey returns a key by ID
```

Comment:
Well, my logic was that CryptoService is supposed to be a SigningService and KeyService:

```go
// CryptoService defines a unified Signing and Key Service as this
// will be most useful for most applications.
type CryptoService interface {
        SigningService
        KeyService
}
```

It seemed weird for a KeyService to only provide a Create method, especially since the description comment says "KeyService provides management of keys locally".

You're right that we could operate on the keystore directly. This would be a bit of a pain though, since notary-signer would have to keep track of CryptoService/KeyStore pairs instead of just keeping track of CryptoServices. I'd rather avoid the busy-work of redoing CryptoServiceIndex to keep track of both, but other than that I don't care much.